### ISSUE TITLE
Compressed multipart uploads using the S3 API lose is_compressed on finalisation

### ISSUE THREAD

**Issue by lukegb** [seaweedfs/seaweedfs#6145]
**Describe the bug**

When uploading large files using the multipart uploads via the S3 API, if the file is compressed on upload then the compression metadata will be lost on the chunks when the multipart upload is finalized, resulting in the file becoming unreadable

**System Setup**
- List the command line to start "weed master", "weed volume", "weed filer", "weed s3", "weed mount".
  - `weed -v=4 filer -defaultReplicaPlacement=000 -encryptVolumeData -iam -iam.ip=127.0.0.1 -iam.port=21015 -localSocket=/run/seaweedfs-filer/filer.sock -master=localhost:21000 -port=21010 -s3 -s3.domainName=redacted.domain.name -s3.localSocket=/run/seaweedfs-filer/s3.sock -s3.port=21012 -webdav -webdav.port=21014`
  - `weed master -defaultReplication=000 -mdir=/var/lib/seaweedfs-master/metadata -port=21000`
  - `weed -v=4 volume -dataCenter=ixn-lon1 -dir=/var/lib/seaweedfs-volume/data -dir.idx=/var/lib/seaweedfs-volume/idx -max=0 -mserver=localhost:21000 -port=21100 -rack=vlx-lukegb1`
- OS version
  - NixOS 24.11
- output of `weed version`
  - version 30GB 3.75 N/A linux amd64
    - (yes, I know this is a bit behind)
- if using filer, show the content of `filer.toml`
  - ```toml
    [leveldb2]
    dir = "/var/lib/seaweedfs-filer/leveldb2"
    enabled = true
    ```

**Expected behavior**
A clear and concise description of what you expected to happen.

Files uploaded using the multipart upload S3 API can be read back once the upload has completed successfully.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

https://github.com/lukegb/seaweedfs/commit/624e2d79ce5a03bb60afd8e1293089dd61c60d52 should fix this, but I couldn't figure out how best to test this, so I haven't opened a PR.

**Comment by chrislusf** [comment]
you can manually test it?

**Comment by lukegb** [comment]
Oh sure; I've manually tested it and it works. I just wasn't sure how best to write a unit test for it.

**Comment by lukegb** [comment]
This was fixed by #6188.
